### ISSUE TITLE
Node.js webworker-threads?

### ISSUE THREAD

**Issue by icodeforlove** [icodeforlove/task.js#20]
At the moment we are using fork to get away with the simulation of a thread.

The [webworker-threads](https://github.com/audreyt/node-webworker-threads) module seems like the perfect module to utilize for workers.

Benchmarks are needed in order to make the switch (#16).

**Comment by icodeforlove** [comment]
Initial tests showed that

- message passing 15% faster
- locking work 2x improvement
- sending large objects 3-5x faster

I don't like that we have use  node-gyp, the plugin also broken native promises for me, and required me to actually wrap resolve/reject in a setTimeout(.., 0).

**Comment by icodeforlove** [comment]
try the `webworker-threads` branch

**Comment by icodeforlove** [comment]
the module tradeoff is actually not that bad if your modules are universal for example

```javascript
	task = task.defaults({
		warmStart: true,
		debug: false,
		globals: {
			json: {
				'awesomeData': yield fs.readFileAsync('./__artists__.json', 'utf8')
			},
			modules: {
				'levenshtein': yield requireBundle('fast-levenshtein')
			}
		},
		initialize: function (globals) {
			Object.keys(globals.modules).forEach(function (name) {
				globals.modules[name] = eval(globals.modules[name]);
			});

			Object.keys(globals.json).forEach(function (name) {
				globals.json[name] = eval(globals.json[name]);
			});

			return globals;
		}
	});
```

this would be an example of giving global access to json and universal node modules

**Comment by icodeforlove** [comment]
There were massive tradeoffs with using this solution. The main one being that you cannot use require.

This is being closed in favor of #32
